util/deephash/deephash.go

```diff
@@ -106,9 +106,18 @@ func (s1 *Sum) xor(s2 Sum) {
 }
 
 func (s Sum) String() string {
+	// Note: if we change this, keep in sync with AppendSum
 	return hex.EncodeToString(s.sum[:])
 }
 
+// AppendSum appends the string encoding of this sum (as returned by the String
+// method) to the provided byte slice and returns the extended buffer.
+func (s Sum) AppendSum(b []byte) []byte {
+	var lb [len(s.sum) * 2]byte
+	hex.Encode(lb[:], s.sum[:])
+	return append(b, lb[:]...)
```

Comment:
Yet another example where `hex.AppendEncode` would be a superior API. See https://github.com/golang/go/issues/53693

Perhaps put a TODO here if that proposal is ever accepted?